controllers/admin/AdminCartsController.php

```diff
@@ -427,8 +427,6 @@ public function ajaxProcessUpdateCustomizationFields()
                             continue;
                         } elseif (!ImageManager::resize($tmp_name, _PS_UPLOAD_DIR_ . $file_name . '_small', (int) Configuration::get('PS_PRODUCT_PICTURE_WIDTH'), (int) Configuration::get('PS_PRODUCT_PICTURE_HEIGHT'))) {
                             $errors[] = $this->trans('An error occurred during the image upload process.', array(), 'Admin.Catalog.Notification');
-                        } elseif (!chmod(_PS_UPLOAD_DIR_ . $file_name, 0777) || !chmod(_PS_UPLOAD_DIR_ . $file_name . '_small', 0777)) {
```

Comment:
It is relevant to remove this processing however can we replace it by an harmless permissions check ? I think this line has been added to handle a bad permissions settings of `_PS_UPLOAD_DIR_`, maybe we can keep the warning ?

Reply:
It is not a warning, it a filemode change, nothing more.

Reply:
I agree with mathieu, it would be nice to keep a check on these permissions, using `fileperms` and bit operations to check the permissions Something like `fileperms($filename) & FileSystem::DEFAULT_MODE == FileSystem::DEFAULT_MODE`

Reply:
There is actually no check and no warning, it is an error.. chmod is not like fileperms.
More, we need to don't force the file change, if someone change is own filesystem mode we mustn't add an error each time he cames here.

Reply:
Yes I understand the difference between `chmod` and `fileperms`, I'm well aware that it wouldn't be the same behavior
But the thing is if the minimum permissions are not set I'm afraid the feature won't work correctly because the image won't be writable, then maybe we should return an explicit error about file permissions so that the user understands that he needs to configure it right

Reply:
We already have this kind of check during installation / upgrade.
If you change the files/directories permissions after because of security or anything else, you don't want to have an error while using an ajax request because of this.
The resize has already been done, so the file is already created.

Reply:
Ok I'm convinced But then I wonder why this chmod was performed before?

Reply:
To force the 0755 mode only :smile: